Arrowgene.Ddon.GameServer/Handler/QuestGetSetQuestListHandler.cs

```diff
@@ -1,32 +1,48 @@
 using Arrowgene.Buffers;
+using Arrowgene.Ddon.GameServer.Characters;
 using Arrowgene.Ddon.GameServer.Dump;
+using Arrowgene.Ddon.GameServer.Quests;
 using Arrowgene.Ddon.Server;
 using Arrowgene.Ddon.Server.Network;
+using Arrowgene.Ddon.Shared.Asset;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
 using Arrowgene.Ddon.Shared.Network;
 using Arrowgene.Logging;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
 
 namespace Arrowgene.Ddon.GameServer.Handler
 {
-    public class QuestGetSetQuestListHandler : PacketHandler<GameClient>
+    public class QuestGetSetQuestListHandler : StructurePacketHandler<GameClient, C2SQuestGetSetQuestListReq>
     {
         private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(QuestGetQuestPartyBonusListHandler));
 
-
         public QuestGetSetQuestListHandler(DdonGameServer server) : base(server)
         {
         }
 
-        public override PacketId Id => PacketId.C2S_QUEST_GET_SET_QUEST_LIST_REQ;
-
-        public override void Handle(GameClient client, IPacket packet)
+        public override void Handle(GameClient client, StructurePacket<C2SQuestGetSetQuestListReq> packet)
         {
-            IBuffer buffer = new StreamBuffer();
-            buffer.WriteUInt32(0);
-            buffer.WriteUInt32(0);
-            buffer.WriteUInt32(0);
-            Packet p = new Packet(PacketId.S2C_QUEST_GET_SET_QUEST_LIST_RES, buffer.GetAllBytes());
-            client.Send(p);
-           // client.Send(GameFull.Dump_132);
+            // client.Send(GameFull.Dump_132);
+
+            S2CQuestGetSetQuestListRes res = new S2CQuestGetSetQuestListRes();
+            foreach (var questId in client.Party.QuestState.GetActiveQuestIds())
+            {
+                var quest = QuestManager.GetQuest(questId);
+                if (quest.QuestType == QuestType.World)
```

Comment:
World Quest Manage, I think these are not actually world quests, but instead the quests which are under the `q7xxxxxxx` quest category. I suspect this because there are quest commands called `WorldManageLayoutxxxx` where the parameters are a flag value and a `q7xxxxxx` questid.

Reply:
What i mean is that someone new looking at the code searching for world quests would look first at world manage quest. I know the terminology used interanlly is different, which is why a comment explaining this context would be good

Reply:
Ok, I added a comment here about this.